test/runtests.jl

```diff
@@ -1,5 +1,50 @@
 using WaveSimulator
+using ComputationalResources
 using Base.Test
 
-# write your own tests here
-#@test 1 == 2
+if Base.Threads.nthreads() > 1
+    addresource(CPUThreads)
+end
+if CUDAdrv.vendor() != "none"
+    addresource(CUDALibs)
+end
+info("Testing resources: $(ComputationalResources.resources)")
+
+# Generate resource instances
+test_resources = AbstractResource[]
+for resource_type in ComputationalResources.resources
+    if resource_type == CPU1
+        push!(test_resources, CPU1())
+    elseif resource_type == CPUThreads
+        push!(test_resources, CPUThreads((64,64,1)))
+    elseif resource_type == CUDALibs
+        push!(test_resources, CUDALibs())
+    end
+end
+
+@testset "Backend Simulation State" begin
+    test_results = Dict()
+    for resource in test_resources
+        # Setup
+        wave = UniformWave{3}(fmax=2e3)
+        sim = Simulator(wave, resource=resource, duration=0.01)
+        domain = BoxDomain(6,8,4, gamma=0.05)
+        f0 = WaveSimulator.gauss(domain)
+        backend = WaveSimulator.backend_init(sim.resource, domain, sim)
+        state = WaveSimulator.state_init(f0, backend, domain, sim)
+        
```

Comment:
I noticed this trailing newline while looking at your PR in VIM. It might be worth looking into your editor settings to see if you can you tweak it to delete trailing newlines automatically. That is a good default behaviour for open source in general.

Reply:
Oh, sorry, that was actually intentional. I tend to put newlines wherever I feel like it helps separate different blocks of code. I'll remove it in my next commit.

Reply:
the newline is completely fine, the white spaces in this line is what my editor flags.